tests/unit/utilities/find-addon-by-name-test.js

```diff
@@ -14,9 +14,17 @@ describe('findAddonByName', function() {
     }, {
       name: 'foo-bar',
       pkg: { name: 'foo-bar' },
+    }, {
+      name: 'foo-bar-baz',
+      pkg: { name: '@foo/bar-baz' },
     }];
   });
 
+  it('should return the foo-bar-baz addon when the unscoped package name is supplied', function() {
+    let addon = findAddonByName(addons, 'bar-baz');
+    expect(addon.name).to.equal('foo-bar-baz', 'should have found the foo-bar-baz addon');
```

Comment:
I am not entirely sure what you are asking. 
The test cases seem to be missing this scenario, where addon.name and addon.pkg.name are both present but have different values. Our package name in package.json is something like "@scope/my-package-name". In index.js the name is "prefix-my-package-name". If I run `ember install @scope/my-package-name` it cannot find the addon, because the name passed in has had the scope removed, while the list of addons has not, and so in the comparison there is no match. (The addon.name is "prefix-my-package-name" and the addon.pkg.name is "@scope/my-package-name", while the base package name passed in is "my-package-name".)